### DISCUSSION TITLE
Upgraded my Intel Mac to an M1 and MySQL under Herd says Bad CPU type

### DISCUSSION THREAD

**Discussion by dstrickler** [beyondcode/herd-community#1161] [Q&A]
Sure, I understand that it's the wrong version - it makes sense.

But how can I re-install MySQL? I've tried downloading Herd again and re-installing it, but that doesn't fix the wrong-version problem. To be clear, I'm not sure how Herd installed MySQL on my Mac, so I don't know how to re-install it.
```
/bin/sh: /Users/Shared/Herd/services/mysql/8.0.36/bin/mysqld: Bad CPU type in executable
/bin/sh: /Users/Shared/Herd/services/mysql/8.0.36/bin/mysqld: Undefined error: 0
```

I have also tried creating multiple MySQL "Services," but none work and I don't know how to remove one.

I'm running the latest Herd and latest MacOS patches.

TIA,

Dave

**Comment by dstrickler** [comment]
Solved: 
1. I Ctrl-Clicked on each service I wanted to delete, and deleted them. This goes for PHP and for MySQL (all three instances I had installed).
2. Once deleted, I could install PHP and MySQL again, but this time Herd downloaded the code and installed it.

Now Herd looks up and running with MySQL. I did loose my data in MySQL, but it was just test data.

**Comment by Mrkbingham** [comment]
I also tried right-clicking in the UI and hitting "delete", but wasn't able to get the same results through this method:

![image](https://github.com/user-attachments/assets/6858a9d2-3883-4a7b-bafe-3415c736e3a6)

But thanks for sharing your insights here @dstrickler - it led me to discover the shared services folder, and once I manually removed those files, re-installing the service worked fine for me!

To delete the shared service (in this case mysql), I removed the following folder:
`/Users/Shared/Herd/services/mysql`

But the same would go for any of the other services like `postgresql`, `reverb`, `redis`, etc.
